src/mlib/endian.h

```diff
@@ -0,0 +1,42 @@
+#ifndef MLIB_ENDIAN_H_INCLUDED
+#define MLIB_ENDIAN_H_INCLUDED
```

Comment:
Possibly. I added this becuase I was going to do more endian handling in int128, but it turns out it wasn't necessary at that level. Checking some pre-defined macros is generally infinitely faster than CMake's "try to build a project for this one small source file to see what it outputs." Of course, we'll still pay for that configure-time check unless we refactor out the existing usage of the configure-time results. There are only nine actual lines in the whole project that depend on `MONGOCRYPT_{BIG,LITTLE}_ENDIAN` (just to do some byte-swaps), so it may be worth it, but perhaps some other time.